File: app/src/main/java/ae/innovativesolutions/wisdomhours/data/AppDataManager.java
package ae.innovativesolutions.wisdomhours.data;

import android.content.Context;

import javax.inject.Inject;
import javax.inject.Singleton;

import ae.innovativesolutions.wisdomhours.data.local.db.DbHelper;
import ae.innovativesolutions.wisdomhours.data.local.prefs.PreferencesHelper;
import ae.innovativesolutions.wisdomhours.data.remote.ApiHelper;

/**
 * Created by Zuhaib Ahmad on 07/07/17.
 */
@Singleton
public class AppDataManager implements DataManager {

    private final Context mContext;
    private final DbHelper mDbHelper;
    private final PreferencesHelper mPreferencesHelper;
    private final ApiHelper mApiHelper;

    @Inject
    public AppDataManager(Context context,
                          DbHelper dbHelper,
                          PreferencesHelper preferencesHelper,
                          ApiHelper apiHelper) {
        mContext = context;
        mDbHelper = dbHelper;
        mPreferencesHelper = preferencesHelper;
        mApiHelper = apiHelper;
    }
}

File: app/src/main/java/com/zuhaibahmad/template/data/AppDataManager.kt
package com.zuhaibahmad.template.data

import android.content.Context
import com.zuhaibahmad.template.data.local.db.DbHelper
import com.zuhaibahmad.template.data.local.prefs.PreferencesHelper
import com.zuhaibahmad.template.data.remote.ApiHelper
import com.zuhaibahmad.template.utils.OpenClass
import javax.inject.Inject
import javax.inject.Singleton


/**
 * Created by Zuhaib on 9/16/2017.
 *
 * A centralized object for management of data throughout application from various sources
 */

@OpenClass
@Singleton

class AppDataManager
@Inject constructor(
	private val context: Context,
	private val dbHelper: DbHelper,
	private val preferencesHelper: PreferencesHelper,
	private val apiHelper: ApiHelper
) : DataManager {

	override val isUserOnline
		get() = apiHelper.isUserOnline

	override var isUserSignedIn = preferencesHelper.isUserSignedIn
}
